pandas/core/base.py

```diff
@@ -1134,8 +1135,20 @@ def factorize(
         self,
         sort: bool = False,
         use_na_sentinel: bool = True,
-    ):
-        return algorithms.factorize(self, sort=sort, use_na_sentinel=use_na_sentinel)
+    ) -> tuple[npt.NDArray[np.intp], Index]:
+
+        codes, uniques = algorithms.factorize(
+            self._values, sort=sort, use_na_sentinel=use_na_sentinel
+        )
+
+        if isinstance(self, ABCIndex):
+            # preserve e.g. NumericIndex, preserve MultiIndex
+            uniques = self._constructor(uniques)
```

Comment:
what's the reason to change this from
```
            uniques = ensure_wrapped_if_datetimelike(uniques)
            uniques = self._shallow_copy(uniques, name=None)
```
? Is it just simpler?

Reply:
correct